x86_64/src/mptable.rs

```diff
@@ -89,6 +96,10 @@ fn compute_mp_size(num_cpus: u8) -> usize {
 
 /// Performs setup of the MP table for the given `num_cpus`.
 pub fn setup_mptable(mem: &GuestMemory, num_cpus: u8) -> Result<()> {
+    if num_cpus as u32 > MAX_SUPPORTED_CPUS {
```

Comment:
Can you also add a unit test for the error case?

Reply:
done